sorc/chgres_cube.fd/surface.F90

```diff
@@ -4072,203 +3538,74 @@ subroutine nst_land_fill
-    call error_handler("IN FieldGet", rc)
-
- where(mask_ptr /= 0) data_ptr = 0.0
-
-! xs
-
- print*,"- CALL FieldGet FOR XS."
- call ESMF_FieldGet(xs_target_grid, &
-                    farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
-    call error_handler("IN FieldGet", rc)
-
- where(mask_ptr /= 0) data_ptr = 0.0
-
-! xt
-
- print*,"- CALL FieldGet FOR XT."
- call ESMF_FieldGet(xt_target_grid, &
-                    farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
-    call error_handler("IN FieldGet", rc)
-
- where(mask_ptr /= 0) data_ptr = 0.0
-
-! xu
-
- print*,"- CALL FieldGet FOR XU."
- call ESMF_FieldGet(xu_target_grid, &
-                    farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
-    call error_handler("IN FieldGet", rc)
-
- where(mask_ptr /= 0) data_ptr = 0.0
-
-! xv
-
- print*,"- CALL FieldGet FOR XV."
- call ESMF_FieldGet(xv_target_grid, &
-                    farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
-    call error_handler("IN FieldGet", rc)
-
- where(mask_ptr /= 0) data_ptr = 0.0
-
 ! xz
 
  print*,"- CALL FieldGet FOR XZ."
  call ESMF_FieldGet(xz_target_grid, &
                     farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
+ if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__))&
     call error_handler("IN FieldGet", rc)
 
  where(mask_ptr /= 0) data_ptr = 30.0
 
-! xtts
-
- print*,"- CALL FieldGet FOR XTTS."
- call ESMF_FieldGet(xtts_target_grid, &
-                    farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
-    call error_handler("IN FieldGet", rc)
-
- where(mask_ptr /= 0) data_ptr = 0.0
-
-! xzts
-
- print*,"- CALL FieldGet FOR XZTS."
- call ESMF_FieldGet(xzts_target_grid, &
-                    farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
-    call error_handler("IN FieldGet", rc)
-
- where(mask_ptr /= 0) data_ptr = 0.0
-
-! z_c
-
- print*,"- CALL FieldGet FOR Z_C."
- call ESMF_FieldGet(z_c_target_grid, &
-                    farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
-    call error_handler("IN FieldGet", rc)
-
- where(mask_ptr /= 0) data_ptr = 0.0
-
-! zm
-
- print*,"- CALL FieldGet FOR ZM."
- call ESMF_FieldGet(zm_target_grid, &
-                    farrayPtr=data_ptr, rc=rc)
- if(ESMF_logFoundError(rcToCheck=rc,msg=ESMF_LOGERR_PASSTHRU,line=__LINE__,file=__FILE__)) &
-    call error_handler("IN FieldGet", rc)
+ do i = 1,16
```

Comment:
There is an ESMF routine that will return the number of fields (fieldCount) within a bundle:
http://earthsystemmodeling.org/docs/release/ESMF_8_1_1/ESMF_refdoc/node5.html#SECTION050251000000000000000

Reply:
Yeah, I use that functionality in the new *****_many routines in the sfc_regrid_many branch, but in this case I know the field number exactly so I didn't really see the benefit to calling another ESMF routine in this case.

Reply:
In that case please define a parameter NUM_ESMF_FIELDS or something of that sort. You should not use bare constants in production code (it's OK to do so in test code).

Reply:
Done. I also converted the other bare constants that had been used in the code prior to my changes to parameters as well.